crates/red_knot_module_resolver/src/path.rs

```diff
@@ -274,23 +364,31 @@ impl<'a> ModuleResolutionPathRefInner<'a> {
 
     #[must_use]
     fn to_module_name(self) -> Option<ModuleName> {
-        let (fs_path, skip_final_part) = match self {
-            Self::Extra(path) | Self::FirstParty(path) | Self::SitePackages(path) => (
-                path,
-                path.ends_with("__init__.py") || path.ends_with("__init__.pyi"),
-            ),
-            Self::StandardLibrary(path) => (path, path.ends_with("__init__.pyi")),
-        };
-
-        let parent_components = fs_path
-            .parent()?
-            .components()
-            .map(|component| component.as_str());
-
-        if skip_final_part {
-            ModuleName::from_components(parent_components)
-        } else {
-            ModuleName::from_components(parent_components.chain(fs_path.file_stem()))
+        match self {
+            Self::Extra(path) | Self::FirstParty(path) | Self::SitePackages(path) => {
+                let parent_components = path
+                    .parent()?
+                    .components()
+                    .map(|component| component.as_str());
+                if path.ends_with("__init__.py") || path.ends_with("__init__.pyi") {
+                    ModuleName::from_components(parent_components)
+                } else {
+                    ModuleName::from_components(parent_components.chain(path.file_stem()))
+                }
+            }
+            Self::StandardLibrary(path) => {
+                let parent = path.parent()?;
+                let parent_components = parent.components().map(|component| component.as_str());
```

Comment:
This is doing exactly the same as in the above clause, why should it be structured differently (which obscures the fact that it's actually the same)?

Reply:
If I make this change:

```diff
--- a/crates/red_knot_module_resolver/src/path.rs
+++ b/crates/red_knot_module_resolver/src/path.rs
@@ -376,8 +376,10 @@ impl<'a> ModuleResolutionPathRefInner<'a> {
                 }
             }
             Self::StandardLibrary(path) => {
-                let parent = path.parent()?;
-                let parent_components = parent.components().map(|component| component.as_str());
+                let parent_components = path
+                    .parent()?
+                    .components()
+                    .map(|component| component.as_str());
```

The borrow checker complains at me:

```
error[E0716]: temporary value dropped while borrowed
   --> crates/red_knot_module_resolver/src/path.rs:379:41
    |
379 |                   let parent_components = path
    |  _________________________________________^
380 | |                     .parent()?
    | |______________________________^ creates a temporary value which is freed while still in use
381 |                       .components()
382 |                       .map(|component| component.as_str());
    |                                                           - temporary value is freed at the end of this statement
...
388 |                       ModuleName::from_components(parent_components)
    |                                                   ----------------- borrow later used here
    |
    = note: consider using a `let` binding to create a longer lived value
```

I can't really understand why it's okay with the first branch but not with the second. I guess I'll make the first branch more like the second.